### PR TITLE
Prevent multiline matrix payload from causing issues in GitHub workflow output

### PR DESCRIPTION
**Pull request by gsteel**
Makes CI green.

It appears that sergeysova/jq-action@v2.3.0 was at fault here, and enabling its `multiline` option solves the issue.

This patch also updates usage of `$GITHUB_OUTPUT` in `continuous-integration.yml` in case the matrix generated there is also multiline.

Deprecations are still being issued about `set-output` but I'm not sure where these are coming from 🤷‍♂️

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -25,7 +25,11 @@ jobs:
         # This provides a dedicated "check" to asynchronously test all integration tests within the tests/ directory
         # The project name from the tests/ directory is then re-used within the "qa" job to run the generated "container"
         # within that directory.
-        run: cd tests; echo "matrix=[\"$(ls -d * | tr '\n' ' ' | sed 's/ $//' | sed 's/ /\",\"/g')\"]" >> $GITHUB_OUTPUT
+        run: |
```
Comment (internalsystemerror) [inline .github/workflows/continuous-integration.yml:28]: Is this needed ASWELL as pinning the version of `jq-action`? I would hope this change would be all that's needed?

Comment (gsteel) [reply .github/workflows/continuous-integration.yml:28]: I don't think so - But, if there's any chance that `matrix` _is_ multiline, at least the case is covered. It should probably always have been like this WRT #150 

We may be able to stick with `jq-action@v2` if the `multiline` option works out for us.

**Review by Ocramius — APPROVED** [review]
Thanks @gsteel!

File: .github/workflows/continuous-integration.yml (hunk 1 above)
Comment (internalsystemerror) [reply .github/workflows/continuous-integration.yml:28]: I think this change needs reverting as `"matrix=[\"$(ls -d * | tr '\n' ' ' | sed 's/ $//' | sed 's/ /\",\"/g')\"]"` should be on a single line.

Comment (gsteel) [reply .github/workflows/continuous-integration.yml:28]: I'll send in another patch @internalsystemerror for 1.23

Comment (internalsystemerror) [reply .github/workflows/continuous-integration.yml:28]: Ty <3

**Pull request merged**